cleverpush/src/main/java/com/cleverpush/BroadcastReceiverHandler.java

```diff
@@ -1,42 +1,35 @@
 package com.cleverpush;
 
-import static com.cleverpush.CleverPush.context;
+import static com.cleverpush.util.BroadcastReceiverUtils.sendBroadcastMessage;
 
 import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
 import android.content.SharedPreferences;
 import android.preference.PreferenceManager;
 
-import com.cleverpush.util.Logger;
-
-public class DeviceIdBroadcastReceiver extends BroadcastReceiver {
+public class BroadcastReceiverHandler extends BroadcastReceiver {
     @Override
     public void onReceive(Context context, Intent intent) {
+        String action = intent.getAction();
+
         if (intent.getStringExtra(Constants.GET_FULL_PACKAGE_NAME_KEY).equals(Constants.APPLICATION_PACKAGE_NAME)) {
```

Comment:
not sure if we need a null check here

Reply:
dont remove  @ozamanav

Reply:
@jprajapati1996

Reply:
@jprajapati1996 

Kevin means we can add *null check* also to avoid errors :-) 

maybe like below
`if (intent.hasExtra(Constants.GET_FULL_PACKAGE_NAME_KEY) && intent.getStringExtra(Constants.GET_FULL_PACKAGE_NAME_KEY).equals(Constants.APPLICATION_PACKAGE_NAME)) {
`

CC: @cpKevinrodriguesnobre

Reply:
okay added
![image](https://user-images.githubusercontent.com/104719696/195089677-848d22c0-481f-4f81-b047-00546bdaa859.png)